src/gentropy/l2g.py

```diff
@@ -82,9 +85,13 @@ def __init__(
                 raise ValueError(
                     "model_path and predictions_path must be set for predict mode."
                 )
-            predictions = L2GPrediction.from_credible_set(
+            predictions, feature_matrix = L2GPrediction.from_credible_set(
                 model_path, list(features_list), credible_set, studies, v2g, coloc
             )
+            if feature_matrix_path:
```

Comment:
I'd remove this conditional block and put it above 
`if not model_path or not predictions_path or not feature_matrix_path`

Reply:
@ireneisdoomed This would mean that we require user to set up the `feature_matrix_path`. My idea was to make it as an optional branch of execution, that we run in case the user provides the flag. Otherwise it will behave as regular - see the default value is None, so default behaviour did not change.

Reply:
Are you trying to cover the scenario where you use the same L2G model and the same data to build the feature to extract predictions on different datasets?

Reply:
(If so, that makes sense :) )